### PR TITLE
[microNPU] Add support for conv2d running on two cores on U65

### PR DESCRIPTION
**Pull request by ekalda**
The 512 mac variant has two cores that processes the weights in
parallel, so we need to split the weights and biases into two
and encode them separately.

cc @Mousius

### CODE REVIEW

**Comment by ekalda** [comment]
cc: @manupa-arm @lhutton1 @NicolaLancellotti @mbaret

**Review by lhutton1 — COMMENTED** [review]
Thanks @ekalda, just small things and a question :)

File: python/tvm/relay/backend/contrib/ethosu/legalize.py [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ def callback(
             )
         else:
             weight_scale = 1 / (filter_height * filter_width)
-            weight_values = np.ones([out_channels, filter_height, filter_width, in_channels])
+            weight_values = np.ones([out_channels, filter_height, filter_width, 1])
```
[the remark this answers is not in the document]
Comment (ekalda) [reply python/tvm/relay/backend/contrib/ethosu/legalize.py:1219]: This is a bug in the mean implementation that I discovered by accident... Since depthwise convolution doesn't contract along the channels axis, one of the dimensions of the 4D weights has to be 1, so e.g. for the (1, 8, 16, 16) IFM it was creating 32 768 values instead of the necessary 2048, the reason we got away with it so far is that the array got flattened and then DivideConstants pass bluntly read only the first 2048 values (it's a tensor of ones, so there was no problem with the output values mismatching either).

File: tests/python/contrib/test_ethosu/test_codegen.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def tf_function(self, x):
                 op = tf.nn.conv2d(
                     x,
                     filters=tf.constant(
-                        np.random.uniform(size=[kernel_shape[0], kernel_shape[1], 3, 3]),
+                        np.random.uniform(size=[kernel_shape[0], kernel_shape[1], ifm_shape[3], 5]),
```
[the remark this answers is not in the document]
Comment (ekalda) [reply tests/python/contrib/test_ethosu/test_codegen.py]: Done!

File: python/tvm/relay/backend/contrib/ethosu/legalize.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lhutton1) [reply python/tvm/relay/backend/contrib/ethosu/legalize.py:1219]: Ah I see, thanks for the fix!

**Review by lhutton1 — APPROVED** [review]
LGTM!

File: tests/python/contrib/test_ethosu/test_replace_conv2d.py [hunk 3]
```diff
@@ -132,6 +132,28 @@ def _create_serial_conv2d_params(
...
 
 
+def get_conv2d_args(call, include_buffers=False, remove_constants=False):
+    """A method to extract the arguments from conv2d extern call."""
+    args = call.args
+    conv_args = []
+    remove_indices = [0]
+
+    if remove_constants:
+        remove_indices += [41, 42, 43, 44, 46, 47, 48, 49]
```
Comment (manupa-arm) [inline tests/python/contrib/test_ethosu/test_replace_conv2d.py:142]: docs : I think we need an explanation about these constants.

**Review by manupa-arm — COMMENTED** [review]
LGTM bar one suggestion.

File: tests/python/contrib/test_ethosu/test_replace_conv2d.py (hunk 3 above)
Comment (ekalda) [reply tests/python/contrib/test_ethosu/test_replace_conv2d.py:142]: Good point - I'll do it in the follow up!
